foundation-ui/mount/Mount.js

```diff
@@ -0,0 +1,108 @@
+import React, {PropTypes} from 'react';
+
+import MountService from './MountService';
+import {CHANGE} from './MountEvent';
+import ReactUtil from '../utils/ReactUtil';
+
+const METHODS_TO_BIND = [
+  'onMountServiceChange'
+];
+
+/**
+ * A component to use as a mount point to incorporate views/components provided
+ * by packages. It renders with a/components registered to the `MountService`
+ * with a matching `type`.
+ * It provides a property to configure the wrapping component (`wrapper`) as
+ * well as a property to `limit` the number of rendered elements.
+ *
+ * @example
+ *
+ * <Mount type="widget" limit={10} wrapper={MyWidgetsWrapper}>
+ *     <span className="help-text"></span>
+ * </Mount>
+ *
+ */
+class Mount extends React.Component {
+  constructor() {
+    super(...arguments);
+
+    // Get components and init state
+    this.state = {
+      components: MountService.findComponentsWithType(this.props.type)
+    };
+
+    METHODS_TO_BIND.forEach((method) => {
+      this[method] = this[method].bind(this);
+    });
+  }
+
+  componentWillMount() {
+    MountService.addListener(CHANGE, this.onMountServiceChange);
+  }
+
+  componentWillReceiveProps({type}) {
+    if (this.props.type === type) {
+      return;
+    }
+
+    this.setState({components: MountService.findComponentsWithType(type)});
+  }
+
+  componentWillUnmount() {
+    MountService.removeListener(CHANGE, this.onMountServiceChange);
+  }
+
+  onMountServiceChange(type) {
+    if (this.props.type !== type) {
+      return;
+    }
+
+    this.setState({components: MountService.findComponentsWithType(type)});
+  }
+
+  render() {
+    const {alwaysWrap, children, limit, wrapper} = this.props;
+    const {components} = this.state;
+
+    // Filter consumed props to only pass on a "clean" set of props
+    const filteredProps = Object.keys(this.props).reduce((props, key) => {
```

Comment:
Won't this prevent a Mount from passing on additional context via props?

I'd imagine we may want to do something like (bad example for dcos-ui but hopefully it illustrates my point):
```
<Mount type="service-table-row" serviceId="/foo">
 <tr>{ services[serviceId] }</tr>
</Mount>
```
So another component that gets injected can consume `serviceId`.